sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -56,6 +64,66 @@ export function findOpinionIndex(pointer: string): OpinionIndex {
   }
 }
 
+const jsEncodingUnit = "Utf16CodeUnit";
+
 export function addStrEncodingParam<T>(options: T): T & { stringIndexType: StringIndexType } {
-  return { ...options, stringIndexType: "Utf16CodeUnit" };
+  return { ...options, stringIndexType: jsEncodingUnit };
+}
+
+export function addEncodingParamToTask<X, Y>(
+  task: X & { parameters?: Y & { stringIndexType?: StringIndexType } }
+): X & { parameters?: Y & { stringIndexType?: StringIndexType } } {
+  if (task.parameters) {
+    task.parameters.stringIndexType = jsEncodingUnit;
+  }
+  return task;
+}
+
+export interface PageParam {
+  top: number;
+  skip: number;
+}
+
+function findParamValue(matches: RegExpMatchArray, param: string): number {
+  for (let i = 0; i < matches.length; i += 2) {
+    if (matches[i] === `\$${param}`) {
+      return parseInt(matches[i + 1]);
+    }
+  }
+  throw new Error(`The parameter \$${param} was not found in nextLink`);
+}
+
+export function nextLinkToTopAndSkip(nextLink: string): PageParam {
```

Comment:
I would prefer instead of us manually regexing URLs that we instead use `URL` and `URLSearchParams`

https://developer.mozilla.org/en-US/docs/Web/API/URL
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
Done.